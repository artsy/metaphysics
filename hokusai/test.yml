version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
