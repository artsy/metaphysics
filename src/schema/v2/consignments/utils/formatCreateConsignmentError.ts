import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
