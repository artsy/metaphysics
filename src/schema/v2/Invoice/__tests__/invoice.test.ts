import gql from "lib/gql"
import { runQuery } from "schema/v2/test/utils"

describe("Invoice", () => {
  it("fetches invoice by token and matches snapshot", async () => {
    const invoicesLoader = jest.fn().mockReturnValue(
      Promise.resolve({
        number: 25173,
        state: "ready",
        ready_at: "2024-10-12T00:00:00Z",
        name: "Percy Z",
        email: "percy.z@example.com",
        line_items: [
          {
            subtotal_cents: 50000,
            description: "Artwork Purchase",
            quantity: 1,
          },
        ],
        payments: [
          {
            successful: true,
            created_at: "2024-10-13T00:00:00Z",
            amount_cents: 50000,
            credit_card: {
              last_digits: "1234",
              brand: "Visa",
            },
          },
        ],
        remaining_cents: 0,
        external_note: "Thank you for your purchase",
        currency: "USD",
      })
    )

    const query = gql`
      {
        invoice(token: "invoice-token") {
          number
          state
          readyAt
          name
          email
          lineItems {
            description
            subtotal
            quantity
          }
          payments {
            successful
            createdAt
            amount
            creditCard {
              lastDigits
              brand
            }
          }
          remaining
          externalNote
          currency
        }
      }
    `

    const { invoice } = await runQuery(query, { invoicesLoader })

    expect(invoice).toMatchSnapshot()
  })
})
