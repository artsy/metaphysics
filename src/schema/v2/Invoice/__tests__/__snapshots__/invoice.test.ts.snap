// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice fetches invoice by token and matches snapshot 1`] = `
{
  "currency": "USD",
  "email": "percy.z@example.com",
  "externalNote": "Thank you for your purchase",
  "lineItems": [
    {
      "description": "Artwork Purchase",
      "quantity": 1,
      "subtotal": "$500",
    },
  ],
  "name": "Percy Z",
  "number": "25173",
  "payments": [
    {
      "amount": "$500",
      "createdAt": "2024-10-13T00:00:00Z",
      "creditCard": {
        "brand": "Visa",
        "lastDigits": "1234",
      },
      "successful": true,
    },
  ],
  "readyAt": "2024-10-12T00:00:00Z",
  "remaining": "$0",
  "state": "READY",
}
`;
